### ISSUE TITLE
Issues on Windows

### ISSUE THREAD

**Issue by KelSolaar** [magnars/multiple-cursors.el#297]
Hi,

The package seems to not work properly on Windows with Spacemacs & Emacs 25.2: It seems like whenever I do for example `SPC SPC mc/edit-beginnings-of-lines ENTER` the command seems to expect something else and does not get issued.

I then press `ESC` and get that message:

> Error in post-command-hook (mc/execute-this-command-for-all-cursors): (quit)

I recorded a screencast to illustrate the problem: https://drive.google.com/open?id=0B_IQZQdc4Vy8UWNkT1VXNWJpbGc

The same Spacemacs configuration works without issues on macOs though.

I'm happy to provide logs, tracebacks assuming you guide me on how to do so.

Cheers,

Thoms

**Comment by nispio** [comment]
To get a traceback, execute `M-x toggle-debug-on-error` and then run your test case again.  Come back and post the contents of the `*Backtrace*` buffer that will appear when the error occurs.

**Comment by KelSolaar** [comment]
So here is the traceback:

```
Debugger entered--Lisp error: (error "Mark a set of lines first")
  signal(error ("Mark a set of lines first"))
  error("Mark a set of lines first")
  mc/edit-lines()
  mc/edit-beginnings-of-lines()
  funcall-interactively(mc/edit-beginnings-of-lines)
  call-interactively(mc/edit-beginnings-of-lines record nil)
  command-execute(mc/edit-beginnings-of-lines record)
  #[0 "\303\301\304\"\210\302\300!\207" ["mc/edit-beginnings-of-lines" mc/edit-beginnings-of-lines #[257 "\211\301\"B\211\207" [extended-command-history delete] 5 "\n\n(fn COMMAND)"] command-execute record] 3 "\n\n(fn)"]()
  funcall(#[0 "\303\301\304\"\210\302\300!\207" ["mc/edit-beginnings-of-lines" mc/edit-beginnings-of-lines #[257 "\211\301\"B\211\207" [extended-command-history delete] 5 "\n\n(fn COMMAND)"] command-execute record] 3 "\n\n(fn)"])
  helm-M-x(nil "mc/edit-beginnings-of-lines")
  funcall-interactively(helm-M-x nil "mc/edit-beginnings-of-lines")
  call-interactively(helm-M-x)
  mc/execute-command(helm-M-x)
  byte-code("\302!\210\303	!\207" [cmd id mc/execute-command mc/create-fake-cursor-at-point] 2)
  mc/execute-command-for-fake-cursor(helm-M-x #<overlay in no buffer>)
  #[(cursor) "\212\302	\")\207" [cmd cursor mc/execute-command-for-fake-cursor] 3](#<overlay in no buffer>)
  mapc(#[(cursor) "\212\302	\")\207" [cmd cursor mc/execute-command-for-fake-cursor] 3] (#<overlay in no buffer> #<overlay from 1 to 2 in %1>))
  mc/execute-command-for-all-fake-cursors(helm-M-x)
  mc/execute-this-command-for-all-cursors-1()
  mc/execute-this-command-for-all-cursors()
```

and the sequence I do to get it: 

- `SPC SPC mc/edit-beginnings-of-lines ENTER`, nothing happens
- then `ENTER` and I get the above traceback.

**Comment by nispio** [comment]
In the traceback you posted, it seems that you do not have a region marked.  Did you highlight a region before running the mc command this time?

**Comment by KelSolaar** [comment]
Yes, as you can see from the screencast, it is selected with mouse though, let me try using keyboard.

**Comment by KelSolaar** [comment]
So performing visual selection with keyboard yields the same behaviour, at least it is consistent from that perspective.

**Comment by nispio** [comment]
And in all cases the error is "Mark a set of lines first"? Because that seems different than the screencast.  Do the symptoms change from those seen in the screencast after you turn on `debug-on-error`?

From the traceback, it looks like you are already in `multiple-cursors-mode` at the time that you invoke `mc/edit-beginning-of-lines`.  That is a problem.  Compounding the issue is the fact that `SPC SPC` appears to call `helm-M-x` which has known problems when being invoked from withing `multiple-cursors-mode`.

Try restarting Spacemacs afresh.  Then run `toggle-debug-on-error`, then use the keyboard to select a region, then run `mc/edit-beginnings-of-lines` and verify whether the traceback matches the one you posted above.  If not, post the new traceback here.  If so, I recommend binding `mc/edit-beginnings-of-lines` to a key in your config, restarting Spacemacs, and trying the whole process again to see if it fixes/changes the issue.

**Comment by nispio** [comment]
Also as a general note, I don't think `multiple-cursors` and Spacemacs with vim keybindings are known to play well together generally.  For example, see this issue #216 and this thread: https://github.com/syl20bnr/spacemacs/issues/2669

**Comment by KelSolaar** [comment]
@nispio : The behaviour is indeed slightly different when I set `toggle-debug-on-error`:

- Without `toggle-debug-on-error` set, the first time I press `SPC SPC mc/edit-beginnings-of-lines ENTER` nothing happens, the Spacemacs *HELM M-x* buffer stays open and repeated `ENTER` key press will clear the cursor markers. Ultimately the *HELM M-x* buffer closes and no cursors are left. `ESC` will generate the error mentioned in OP.
- With `toggle-debug-on-error` set, I get the traceback above and the *HELM M-x* buffer closes immediately.

I will read the above mentioned issues, oddly enough though this is a Windows only issue in my case.

**Comment by KelSolaar** [comment]
I was wondering that if I actually needed to enable a *Spacemacs* layer to have that working better.
